src/rest-server/src/models/v2/utils/frameworkConverter.js

```diff
@@ -128,25 +120,95 @@ const generateExitDiagnostics = (diag) => {
   return exitDiagnostics;
 };
 
+const generateSpecMap = () => {
+  let exitSpecPath;
+  if (process.env[env.exitSpecPath]) {
+    exitSpecPath = process.env[env.exitSpecPath];
+    if (!path.isAbsolute(exitSpecPath)) {
+      exitSpecPath = path.resolve(__dirname, '../../', exitSpecPath);
+    }
+  } else {
+    exitSpecPath = '/k8s-job-exit-spec-configuration/k8s-job-exit-spec.yaml';
+  }
+  const exitSpecList = yaml.safeLoad(fs.readFileSync(exitSpecPath));
+  const exitSpecMap = {};
+  exitSpecList.forEach((val) => {
+    exitSpecMap[val.code] = val;
+  });
+
+  return exitSpecMap;
+};
+
+const generateExitSpec = (code) => {
+  const exitSpecMap = generateSpecMap();
+  if (!_.isNil(code)) {
+    if (!_.isNil(exitSpecMap[code])) {
+      return exitSpecMap[code];
+    } else {
+      if (code > 0) {
+        return {
+          ...exitSpecMap[positiveFallbackExitCode],
+          code,
+        };
+      } else {
+        return {
+          ...exitSpecMap[negativeFallbackExitCode],
+          code,
+        };
+      }
+    }
+  } else {
+    return null;
+  }
+};
+
 const convertState = (state, exitCode) => {
   switch (state) {
```

Comment:
Copy this convertState to replace convertFrameworkState  in \database-controller\src\common\framework.js

Reply:
> Copy this convertState to replace convertFrameworkState in \database-controller\src\common\framework.js

Replaced, thanks